cwms-data-api/src/test/java/cwms/cda/data/dao/basin/BasinDaoTest.java

```diff
@@ -0,0 +1,343 @@
+                Basin retrievedBasin = basinDao.getBasin(renamedBasin.getBasinId(), UNIT_SYSTEM);
+                assertNotNull(retrievedBasin);
+                assertSame(renamedBasin, retrievedBasin);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+        cleanUpRoutine(renamedBasin);
+    }
+
+    @Test
+    void testDeleteBasin() throws Exception {
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            try {
+                Basin basin = buildTestBasin(6);
+                assertNotNull(basin);
+                BasinDao basinDao = new BasinDao(ctx);
+                basinDao.storeBasin(basin);
+                basinDao.deleteBasin(basin.getBasinId(), DELETE_ACTION);
+                CwmsId basinId = basin.getBasinId();
+                assertThrows(NotFoundException.class, () -> basinDao.getBasin(basinId, UNIT_SYSTEM));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    private void cleanUpRoutine(Basin basin) throws Exception {
+        assertNotNull(basin);
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c -> {
+            DSLContext ctx = getDslContext(c, OFFICE_ID);
+            try {
+                BasinDao basinDao = new BasinDao(ctx);
+                basinDao.deleteBasin(basin.getBasinId(), DELETE_ACTION);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        });
+    }
+
+    private Basin buildTestBasin(int num) {
+        return new Basin.Builder()
+                .withBasinId(new CwmsId.Builder()
+                        .withName("TEST_BASIN" + num)
+                        .withOfficeId(OFFICE_ID)
+                        .build())
+                .withAreaUnit(UNITS)
+                .withTotalDrainageArea(100.0)
+                .withParentBasinId(parentBasin.getBasinId())
+                .withSortOrder(1.0)
+                .withContributingDrainageArea(50.0)
+                .withPrimaryStreamId(new CwmsId.Builder()
+                        .withName("TEST_PAR_PRIM_STREAM")
+                        .withOfficeId(OFFICE_ID)
+                        .build())
+                .build();
+    }
+
+    private static void assertSame(Basin first, Basin second){
+        assertAll(
+                () -> {
+                    if (first.getBasinId() != null) {
+                        assertEquals(first.getBasinId().getName(), second.getBasinId().getName(), "Basin IDs are not equal");
+                        assertEquals(first.getBasinId().getOfficeId(), second.getBasinId().getOfficeId(), "Basin Offices are not equal");
+                    }
+                },
+                () -> {
```

Comment:
for comparing the streams use StreamTest.assertSame(...)

Reply:
See above comment about removing this assertSame altogether and instead use one defined in BasinTest

Reply:
Fixed in latest commit